synapse/handlers/room.py

```diff
@@ -1271,6 +1260,12 @@ async def create_event(
             )
             events_to_send.append((encryption_event, encryption_context))
 
+        # update event contexts with state group information
+        state = self.hs.get_state_handler()
+        events_to_send = await state.compute_event_context_for_batched(
+            events_to_send, current_state_group, state_map
```

Comment:
`state_map` here has had all the batched events applied to it. But I think the function wants the state map \*before* the whole batch?